pkgs/stdenv/generic/setup.sh

```diff
@@ -100,10 +138,13 @@ _callImplicitHook() {
     local def="$1"
     local hookName="$2"
     if declare -F "$hookName" > /dev/null; then
+        echo "calling implicit '$hookName' function hook"
         "$hookName"
     elif type -p "$hookName" > /dev/null; then
+        echo "sourcing implicit '$hookName' script hook"
         source "$hookName"
     elif [ -n "${!hookName:-}" ]; then
+        echo "evaling implicit '$hookName' string hook"
```

Comment:
For instance, here you don't do the trick with `"${NIX_DEBUG:-0}" >= 2` and just splat it out.

Reply:
Actually this part doesn't splat out the hook string — since implicit hooks are for specifying "one-off" hooks right in a derivation, they're going to be specified by the derivation being built itself. Those often *are* multiline, but since they're also by far the most obvious hooks, so I went with just logging the name. So something like

```nix
preConfigure = ''
  some foo commands
'';

```

Gets logged as `evaling implicit 'preConfigure' hook` without printing the text itself. If you have something better I'm absolutely all ears 🙂

Reply:
That's definitely good enough. Thanks for the example.

I wonder (and this isn't for this review) whether "implicit" is the right name for this concept, and whether "immediate" or "derivation-supplied" might be a better name. In fact, _from the perspective of the derivation or package author_, they might be thought of as "explicit" hooks as opposed to the "implicit" ones that come from elsewhere. An irony if so.

Reply:
A lot of my time spent debugging this ended up actually amounting to figuring out what implicit hooks *are* and where they come from and why there's a separate, special mechanism for them in stdenv 😅

So I completely agree; a rename of the concept would be very helpful. "Immediate" is decent, but further bikeshedding can be at a later date